test/k8sT/Services.go

```diff
@@ -352,6 +352,46 @@ var _ = Describe("K8sServicesTest", func() {
 				testCurlFromPods(echoPodLabel, url, 10, 0)
 			})
 		})
+
+		It("Checks service.kubernetes.io/service-proxy-name label implementation", func() {
+			serviceProxyLabelName := "service.kubernetes.io/service-proxy-name"
+
+			ciliumPods, err := kubectl.GetCiliumPods(helpers.CiliumNamespace)
+			Expect(err).To(BeNil(), "Cannot get cilium pods")
+
+			clusterIP, _, err := kubectl.GetServiceHostPort(helpers.DefaultNamespace, echoServiceName)
+			Expect(err).Should(BeNil(), "Cannot get service %q ClusterIP", echoServiceName)
+			Expect(govalidator.IsIP(clusterIP)).Should(BeTrue(), "ClusterIP is not an IP")
+
+			url := fmt.Sprintf("http://%s/", clusterIP)
+
+			By("Labelling echo service with dummy service-proxy-name")
+			res := kubectl.Exec(fmt.Sprintf("kubectl label services/%s %s=%s", echoServiceName, serviceProxyLabelName, "dummy-lb"))
+			res.ExpectSuccess("cannot label service")
+
+			// Give time for the change in labels to be picked up by the controller.
+			time.Sleep(3 * time.Second)
```

Comment:
This will probably cause flakes. Please refactor following check to use `Eventually`.

Reply:
I did now know about `Eventually`, thanks for pointing it out. I will refactor the tests.